ark/utils/segmentation_utils.py

```diff
@@ -176,3 +176,55 @@ def concatenate_csv(base_dir, csv_files, column_name="point", column_values=None
             combined_data = pd.concat((combined_data, temp_data), axis=0, ignore_index=True)
 
     combined_data.to_csv(os.path.join(base_dir, "combined_data.csv"), index=False)
+
+
+def visualize_segmentation(segmentation_labels_xr, channel_data_xr,
+                           output_dir,
+                           overlay_channels, fovs=None,
+                           save_tifs='overlays'):
+    """Runs the watershed transform over a set of probability masks output by deepcell network
+    Saves xarray to output directory
+
+    Args:
+        segmentation_labels_xr (xarray.DataArray): xarray containing segmentation labels
+        channel_data_xr (xarray.DataArray): xarray containing TIFs
+        output_dir (str): path to directory where the output will be saved
+        overlay_channels (tuple): channels to overlay segmentation output over
+        fovs (numpy.ndarray): field of view
+        save_tifs (str): flag to control what level of output to save.
+            Must be one of:
+
+            * 'all'
+            * 'none'
+            * 'overlays' (saves color overlays and segmentation masks)
+    """
+
+    if fovs is None:
+        fovs = segmentation_labels_xr.fovs
+    for fov in fovs:
+        if save_tifs == 'none':
+            # this should actually plot the images, just not save them to tifs
+            plot_utils.plot_overlay(
+                labels,
+                plotting_tif=input_data,
+            )
+            return
+
+        labels = segmentation_labels_xr.loc[fov, :, :, 'whole_cell'].values
+
+        for chan_list in overlay_channels:
+            input_data = channel_data_xr.loc[fov, :, :, chan_list].values
+            save_path = '_'.join([f'{fov}', *chan_list.astype('str'), 'overlay.tiff'])
+            plot_utils.plot_overlay(
+                labels,
+                plotting_tif=input_data,
+                path=os.path.join(output_dir, save_path)
+            )
+
+            if save_tifs == 'all':
+                plot_utils.plot_overlay(
+                    labels,
+                    plotting_tif=input_data[:, :, 0],
+                    path=os.path.join(output_dir, f'{fov}_segmentation_borders.tiff')
```

Comment:
If you look at the [original function](https://github.com/angelolab/ark-analysis/blob/b9161da0d7041eabbab7f21de089fbc6d3101bd4/segmentation/utils/segmentation_utils.py#L152), we don't supply a plotting tif when plotting the borders. That defeats the point of plotting the borders; then it would be an overlay